Update Kokkos installation instruction (#1256)
PennyLaneAI/pennylane-lightning: doc/lightning_kokkos/installation_hpc.rst

Building and Running Lightning-Kokkos with MPI on Frontier
==========================================================

Here we demonstrate the steps to build Lightning-Kokkos with MPI on `Frontier <https://www.olcf.ornl.gov/frontier/>`_ for AMD GPUs.

We can load the following modules to enable the relevant compilers and Python environment:

.. code-block:: console

    # Load the required Python and compiler modules
    module load cray-python
    module load PrgEnv-amd

Install Kokkos (Recommended)
^^^^^^^^^^^^^^^^^^^^^^^^^

.. note::

    Lightning-Kokkos was tested with Kokkos version <= 4.5.00
    

> We suggest first installing Kokkos with the wanted configuration following the instructions found in the Kokkos documentation. For example, the following will build Kokkos for AMD MI210/250/250X GPUs.

Reviewer: ```suggestion
We suggest first installing Kokkos with the desired configuration, following the instructions found in the Kokkos documentation.
For example, the following instructions demonstrate building Kokkos for AMD MI210/250/250X GPUs:
```

Download the `Kokkos code <https://github.com/kokkos/kokkos/releases>`_.

.. code-block:: bash

    # Replace x, y, and z by the correct version
    wget https://github.com/kokkos/kokkos/archive/refs/tags/4.x.yz.tar.gz
    tar -xvf 4.x.y.z.tar.gz
    cd kokkos-4.x.y.z

Build Kokkos for AMD GPU (``GFX90A`` architecture), and append the install location to ``CMAKE_PREFIX_PATH``.

> .. code-block:: console

Reviewer 2: These are rendering a bit differently than the non-HPC version which use `bash` here

> .. code-block:: console

Author: Good catch, thanks

Building and Running Lightning-Kokkos with MPI on Frontier
==========================================================

Here we demonstrate the steps to build Lightning-Kokkos with MPI on `Frontier <https://www.olcf.ornl.gov/frontier/>`_ for AMD GPUs.

We can load the following modules to enable the relevant compilers and Python environment:

.. code-block:: console

    # Load the required Python and compiler modules
    module load cray-python
    module load PrgEnv-amd

Install Kokkos (Recommended)
^^^^^^^^^^^^^^^^^^^^^^^^^

.. note::

>     Lightning-Kokkos was tested with Kokkos version <= 4.5.00

Author: ```suggestion
    Lightning-Kokkos is tested with Kokkos version <= 4.5.00
```

        -DKokkos_ENABLE_SERIAL:BOOL=ON \
        -DKokkos_ENABLE_HIP:BOOL=ON \
        -DKokkos_ARCH_AMD_GFX90A:BOOL=ON \
        -DKokkos_ENABLE_COMPLEX_ALIGN:BOOL=OFF \
        -DKokkos_ENABLE_EXAMPLES:BOOL=OFF \
        -DKokkos_ENABLE_TESTS:BOOL=OFF \
        -DKokkos_ENABLE_LIBDL:BOOL=OFF
    cmake --build build && cmake --install build
    export CMAKE_PREFIX_PATH=$KOKKOS_INSTALL_PATH  

Install Lightning-Kokkos
^^^^^^^^^^^^^^^^^^^^^^^^

> Lightning-Qubit needs to be 'installed' by ``pip`` before Lightning-Kokkos (compilation is not necessary).

Author: ```suggestion
Lightning-Qubit needs to be 'installed' by ``pip`` before Lightning-Kokkos. This can be installed from source as follows.
```

> Lightning-Qubit needs to be 'installed' by ``pip`` before Lightning-Kokkos. This can be installed from source as follows.

Reviewer: Why 'installed'?
```suggestion
Before Lightning-Kokkos can be installed, Lightning-Qubit needs to be installed via ``pip``.
It can be installed from source as follows:
```

> .. code-block:: console

Reviewer 2: ```suggestion
.. code-block:: bash
```
